sexp/src/test/java/com/novoda/sexp/finder/BasicElementFinderShould.java

```diff
@@ -67,4 +67,35 @@ public void allow_null_results_when_get_result_is_used() throws Exception {
         assertThat(result).isNull();
     }
 
+    @Test
+    public void pop_a_result_when_parsing_finished() throws Exception {
+        String result = "result";
+
+        elementCreator.onParsed(result);
+
+        assertThat(elementCreator.popResult()).isEqualTo(result);
+    }
+
+    @Test(expected = BasicElementFinder.ResultNotFoundException.class)
+    public void throw_exception_when_result_has_not_been_parsed_and_a_result_is_popped() throws Exception {
+        elementCreator.popResultOrThrow();
+    }
+
+    @Test
+    public void allow_null_results_when_get_result_is_popped() throws Exception {
+        Object result = elementCreator.popResult();
+
+        assertThat(result).isNull();
+    }
+
+    @Test
+    public void not_cache_result_after_a_result_is_popped() throws Exception {
+        String result = "result";
+
+        elementCreator.onParsed(result);
+
+        assertThat(elementCreator.popResult()).isEqualTo(result);
```

Comment:
but that's part of the contract and I think it makes it more explicit, it shows that the pop actually works cause  the 1st time you get the result and 2nd time you get null, I think it makes it more explanatory